samza-test/src/main/java/org/apache/samza/test/framework/stream/CollectionStream.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.test.framework.stream;
+
+import com.google.common.base.Preconditions;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.samza.test.framework.system.CollectionStreamSystem;
+
+/**
+ * A CollectionStream represents an in memory stream of messages that can either have single or multiple partitions.
+ * Every CollectionStream is coupled with a {@link CollectionStreamSystem} that
```

Comment:
Does CollectionStream need to be coupled with CollectionStreamSystem? For example, it seems like you could use CollectionStream with some other implementation of a system, such as kafka-based system.

Reply:
Yes a CollectionStream is always associated with an CollectionStreamSystem, it cannot work with like a KafkaSystem